Sources/Purchasing/Purchases/TransactionPoster.swift

```diff
@@ -106,7 +106,10 @@ final class TransactionPoster: TransactionPosterType {
             switch result {
             case .success(let encodedReceipt):
                 self.product(with: productIdentifier) { product in
-                    self.transactionFetcher.appTransactionJWS { appTransaction in
+                    // Only allow refreshing the AppTransaction if the source was a user action (e.g. purchase)
+                    let refreshPolicy: AppTransactionRefreshPolicy =
+                        data.source.initiationSource == .purchase ? .onlyIfEmpty : .never
+                    self.transactionFetcher.appTransactionJWS(refreshPolicy: refreshPolicy) { appTransaction in
```

Comment:
We dont want to potentially show the login prompt for transactions coming from the queue